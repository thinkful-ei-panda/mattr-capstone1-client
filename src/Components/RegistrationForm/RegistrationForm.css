.registration-form {
    width: 90%;
  }

@media screen and (min-width: 1200px) {
    
  
  }