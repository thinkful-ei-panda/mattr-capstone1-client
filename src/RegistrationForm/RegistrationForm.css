body {
  background-size: cover;
  background-image: url("american-flag-small.jpg");
  background-repeat: no-repeat;
  margin-top: 0;
  margin-bottom: 0;
}

Form {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-family: 'Kanit', sans-serif;
  border: 2px solid #9b0202af;
  background-color: #fffffff6;
  
  width: 80%;
  margin-left: auto;
  margin-right: auto;
}

h1 {
  font-family: 'Kanit', sans-serif;
  text-align: center;
  color: white;
  margin-bottom: 0;
}

input {
  margin: 10px;
  /* align-self: center; */
}

.registration__hint {
  font-size: 0.8rem;
}

img {
  height: 100%;
}

.error {
  color: red;
  font-size: 0.7em;
  position: relative;
  
}

@media screen and (min-width: 800px) {
  body {
  background-image: url("american-flag-tablet.jpg");
  background-repeat: no-repeat;
  }
}

@media screen and (min-width: 1200px) {
  body {
  background-image: url("American\ Flag.jpg");
  background-repeat: no-repeat;
  }
}