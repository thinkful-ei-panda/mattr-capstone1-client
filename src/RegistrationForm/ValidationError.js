import React from 'react';
import './Form.css';

export default function ValidationError(props) {
  if(props.message) {
    return (
      <div className="error">{props.message}</div>
    );
  }

  return <></>
}