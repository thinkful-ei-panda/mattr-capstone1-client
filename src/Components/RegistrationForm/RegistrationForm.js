import React, { Component } from "react";
import history from "../../history";
import AuthApiService from "../../Services/auth-api-service";
import "./RegistrationForm.css";
import ValidationError from "../ValidationError";
import Nav from "../Nav/Nav";

const validator = require("email-validator");

export default class RegistrationForm extends Component {
  constructor(props) {
    super(props);
    this.state = {
      email: {
        value: "",
        touched: false,
      },
      password: {
        value: "",
        touched: false,
      },
      repeatPassword: {
        value: "",
        touched: false,
      },
      error: null,
    };
  }

  handleRegistrationSubmit = (ev) => {
    ev.preventDefault();
    const { email, password } = ev.target;

    this.setState({ error: null });
    AuthApiService.postUser({
      user_email: email.value,
      user_password: password.value,
    })
      .then((user) => {
        email.value = "";
        password.value = "";
        this.props.onRegistrationSuccess();
      })
      .catch((res) => {
        this.setState({ error: res.error });
      });
    history.push("/Login");
  };

  static defaultProps = {
    onRegistrationSuccess: () => {},
  };

  updateEmail(email) {
    this.setState({ email: { value: email, touched: true } });
  }

  updatePassword(password) {
    this.setState({ password: { value: password, touched: true } });
  }

  updateRepeatPassword(repeatPassword) {
    this.setState({ repeatPassword: { value: repeatPassword, touched: true } });
  }

  validateEmail() {
    const email = this.state.email.value.trim();
    if (!validator.validate(email)) {
      return "Must enter a valid email";
    }

    return false;
  }

  validatePassword() {
    const password = this.state.password.value.trim();
    if (password.length === 0) {
      return "Password is required";
    } else if (password.length < 6 || password.length > 72) {
      return "Password must be between 6 and 72 characters long";
    } else if (!password.match(/[0-9]/)) {
      return "Password must contain at least one number";
    } else {
      return false;
    }
  }

  validateRepeatPassword() {
    const repeatPassword = this.state.repeatPassword.value.trim();
    const password = this.state.password.value.trim();

    if (repeatPassword !== password) {
      return "Passwords do not match";
    }
    return false;
  }

  render() {
    const emailError = this.validateEmail();
    const passwordError = this.validatePassword();
    const repeatPasswordError = this.validateRepeatPassword();

    return (
      <div className='registration-form'>
        <Nav />
        <div className="login-box animate__animated animate__fadeInDownBig">
          <h2>Register</h2>
          <form onSubmit={this.handleRegistrationSubmit}>
            <div className="registration__hint">* required field</div>
            <div className="user-box">
              <label htmlFor="email">E-mail *</label>
              <input
                type="email"
                className="registration__control"
                name="email"
                id="email"
                onChange={(e) => this.updateEmail(e.target.value)}
              />
              {this.state.email.touched && (
                <ValidationError message={emailError} />
              )}
            </div>

            <div className="user-box">
              <label htmlFor="password">Password *</label>
              <input
                type="password"
                name="password"
                id="password"
                onChange={(e) => this.updatePassword(e.target.value)}
              />
              {this.state.password.touched && (
                <ValidationError message={passwordError} />
              )}
            </div>

            <div className="user-box">
              <label htmlFor="repeatPassword">Repeat Password *</label>
              <input
                type="password"
                name="repeatPassword"
                id="repeatPassword"
                onChange={(e) => this.updateRepeatPassword(e.target.value)}
              />
              {this.state.repeatPassword.touched && (
                <ValidationError message={repeatPasswordError} />
              )}
            </div>

            <div className="vote-link-button-group">
              <button
                type="submit"
                className="vote-link"
                disabled={
                  this.validateEmail() &&
                  this.validatePassword() &&
                  this.validateRepeatPassword()
                }
              >
                <span></span>
                <span></span>
                <span></span>
                <span></span>
                Save
              </button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
