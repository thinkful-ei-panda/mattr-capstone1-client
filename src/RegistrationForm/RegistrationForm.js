import React, { Component } from 'react';
import './RegistrationForm.css';
import ValidationError from './ValidationError';
const validator = require("email-validator");

export default class Form extends Component {
    constructor(props) {
        super(props);
        this.state = {
        firstName: {
            value: '',
            touched: false
        },
        lastName: {
            value: '',
            touched: false
        },
        email: {
            value: '',
            touched: false
        },
        dob: {
            value: '',
            touched: false
        },
        password: {
            value: '',
            touched: false
        },
        repeatPassword: {
            value: '',
            touched: false
        }
      }
    }

    updateFirstName(firstName) {
          this.setState({firstName: {value: firstName, touched: true}})
      };

    updateLastName(lastName) {
        this.setState({lastName: {value: lastName, touched: true}})
    };

    updateEmail(email) {
        this.setState({email: {value: email, touched: true}})
    };

    updateDOB(dob) {
        this.setState({dob: {value: dob, touched: true}})
    };

    updatePassword(password) {
        this.setState({password: {value: password, touched: true}})
    };
      
    updateRepeatPassword(repeatPassword) {
        this.setState({repeatPassword: {value: repeatPassword, touched: true}})
    };

    handleSubmit(event){
        event.preventDefault();
        const { firstName, lastName, email, dob, password, repeatPassword } = this.state;

        console.log('first name: ', firstName.value);
        console.log('last name: ', lastName.value);
        console.log('email: ', email.value);
        console.log('dob: ', dob.value);
        console.log('Password: ', password.value);
        console.log('Repeat Password: ', repeatPassword.value);
    }

    validateFirstName() {
        const firstName = this.state.firstName.value.trim();
        if (firstName.length === 0) {
          return 'Name is required';
        } else if (firstName.length < 3) {
          return 'Name must be at least 3 characters long';
        }
      }

    validateLastName(){
          const lastName = this.state.lastName.value.trim();
          if (lastName.length === 0) {
              return 'Name is required';
          } else if (lastName.length < 2) {
              return 'Name must be at least 3 characters long';
          }
      }

      validateEmail(){
          const email = this.state.email.value.trim();
          if(!validator.validate(email)) {
              return 'Must enter a valid email';
          }
      }

      validatePassword() {
        const password = this.state.password.value.trim();
        if (password.length === 0) {
          return 'Password is required';
        } else if (password.length < 6 || password.length > 72) {
          return 'Password must be between 6 and 72 characters long';
        } else if (!password.match(/[0-9]/)) {
          return 'Password must contain at least one number';
        }
      }

      validateRepeatPassword() {
        const repeatPassword = this.state.repeatPassword.value.trim();
        const password = this.state.password.value.trim();
    
        if (repeatPassword !== password) {
          return 'Passwords do not match';
        }
      }

    render() {
    const firstNameError = this.validateFirstName();
    const lastNameError = this.validateLastName();
    const emailError = this.validateEmail();
    const passwordError = this.validatePassword();
    const repeatPasswordError = this.validateRepeatPassword();

    return (
         <div>
            <form className="registration" onSubmit={e => this.handleSubmit(e)}>
                <h2>Register</h2>
                <div className="registration__hint">* required field</div>  
                <div className="form-group">
                <label htmlFor="first-name">First Name *</label>
                <input type="text" className="registration__control" name="first-name" id="first-name" onChange={e => this.updateFirstName(e.target.value)} defaultValue="first name" />
                <ValidationError message={firstNameError}/>
                {this.state.firstName.touched && <ValidationError message={this.validateFirstName} />}
                </div>
                <div className="form-group">
                <label htmlFor="last-name">Last Name *</label>
                <input type="text" className="registration__control" name="last-name" id="name" onChange={e => this.updateLastName(e.target.value)} defaultValue="last name" />
                {this.state.firstName.touched && <ValidationError message={lastNameError} />}
                </div>
                <div className="form-group">
                <label htmlFor="email">E-mail *</label>
                <input type="email" className="registration__control" name="email" id="email" onChange={e => this.updateEmail(e.target.value)} defaultValue="email address"/>
                {this.state.firstName.touched && <ValidationError message={emailError} />}
                </div>
                <div className="form-group">
                <label htmlFor="dob">Date of Birth *</label>
                <input type="date" className="registration__control" name="dob" id="dob" onChange={e => this.updateDOB(e.target.value)} />
                {/* {this.state.firstName.touched && <ValidationError message={this.validateDOB} />} */}
       </div>

       <div className="form-group">
          <label htmlFor="password">Password *</label>
          <input type="password" className="registration__control" name="password" id="password" onChange = {e => this.updatePassword(e.target.value)} />
          {this.state.firstName.touched && <ValidationError message={passwordError} />}
          <div className="registration__hint">6 to 72 characters, must include a number</div>
       </div>

       <div className="form-group">
         <label htmlFor="repeatPassword">Repeat Password *</label>
         <input type="password" className="registration__control" name="repeatPassword" id="repeatPassword" onChange = {e => this.updateRepeatPassword(e.target.value)} />
         {this.state.firstName.touched && <ValidationError message={repeatPasswordError} />}
       </div>

       <div className="registration__button__group">
        <button type="reset" className="registration__button"> Cancel </button>
        <button type="submit" className="registration__button" disabled={ this.validateFirstName() || this.validateLastName() || this.validateEmail() || this.validatePassword() || this.validateRepeatPassword() } > Save </button>
       </div>
     </form>
        </div>
    )}
}
