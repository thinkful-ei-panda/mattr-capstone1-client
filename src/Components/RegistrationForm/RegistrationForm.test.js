import React from 'react';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import RegistrationForm from './RegistrationForm'
import { it } from 'date-fns/locale';

describe(`RegistrationForm component`, () => {
    const props = {
        className: 'test-class-name',
        'data-other': 'test-other-prop'
    }

    it('renders the RegistrationForm given props', () => {
        const wrapper = shallow(<RegistrationForm {...props} />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})