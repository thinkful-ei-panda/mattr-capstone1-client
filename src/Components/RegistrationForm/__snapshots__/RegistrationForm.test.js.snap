// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RegistrationForm Component Snapshot test Renders the UI as expected 1`] = `
<div>
  <div
    className="Nav"
  >
    <a
      href="/Home"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      href="/Election"
      onClick={[Function]}
    >
      Elections
    </a>
    <a
      href="/RegistrationForm"
      onClick={[Function]}
    >
       
      Register
       
    </a>
    <a
      href="/Login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
  <div
    className="login-box"
  >
    <h2>
      Register
    </h2>
    <form
      onSubmit={[Function]}
    >
      <div
        className="registration__hint"
      >
        * required field
      </div>
      <div
        className="user-box"
      >
        <label
          htmlFor="email"
        >
          E-mail *
        </label>
        <input
          className="registration__control"
          id="email"
          name="email"
          onChange={[Function]}
          type="email"
        />
      </div>
      <div
        className="user-box"
      >
        <label
          htmlFor="password"
        >
          Password *
        </label>
        <input
          id="password"
          name="password"
          onChange={[Function]}
          type="password"
        />
      </div>
      <div
        className="user-box"
      >
        <label
          htmlFor="repeatPassword"
        >
          Repeat Password *
        </label>
        <input
          id="repeatPassword"
          name="repeatPassword"
          onChange={[Function]}
          type="password"
        />
      </div>
      <div
        className="vote-link-button-group"
      >
        <button
          className="vote-link"
          disabled={false}
          type="submit"
        >
          <span />
          <span />
          <span />
          <span />
          Save
        </button>
      </div>
    </form>
  </div>
</div>
`;
